### DISCUSSION TITLE
How to expand multiple relation field

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [pocketbase/pocketbase#3712] [Q&A] [answered]
I have a "posts" collection that has a single relation to an author from the "users" collection. I have another "ratings" collection which stores all the ratings a user has received. My question is how to expand the ratings field in the "author" collection while getting the posts. 

Thats the relation fields I have:

The "posts" collection:
![Screenshot 2023-11-08 at 19 16 50](https://github.com/pocketbase/pocketbase/assets/49731322/f4a14a11-0b9d-4b40-8484-aa65aa14fa13)

The "users" collection:
![Screenshot 2023-11-08 at 19 16 14](https://github.com/pocketbase/pocketbase/assets/49731322/e0ce996c-c21c-4c74-af2e-cae35a64a6f4)

The "ratings" collection:
![Screenshot 2023-11-08 at 19 16 32](https://github.com/pocketbase/pocketbase/assets/49731322/2e1b9130-ff93-49ea-945f-4d42fa5c308a)

```js
await event.locals.pb.collection('posts').getList(page, limit, {
        sort: '-created',
        expand: 'author, author.ratings() //figuring out what should go here,
    });
```

**Comment by ganigeorgiev** [comment]
Using `author.ratings` should work as long as the request satisfies the author and ratings related collections View API rules.

```js
await event.locals.pb.collection('posts').getList(page, limit, {
    sort:   '-created',
    expand: 'author.ratings',
});
```

**Reply by dimitarnikolovv** [reply]
That's what I thought too, but in fact, the ratings are not expanded and I get this data: 

```json
"posts": [
    {
      ...
      "author": "lx11b6foejtfjtd",
      ...
      "expand": {
        "author": {
          ...
          "ratings": [
            "7d1v371d3jdope4",
            "ijuh566gyj9hror",
            "aopnxniz1ybac72"
          ],
        },
      }
    }
]
```

**Reply by ganigeorgiev** [reply]
Make sure that the request satisfies the View API rule of the "ratings" related collection.

Also note that the "ratings" expand will be under a nested "expand" field in the "author" object.

**Reply by dimitarnikolovv** [reply]
Yes I know, that's exactly what I am missing. The "ratings" collection has no rules set for List/Search and View, it is open to everyone.

**Reply by ganigeorgiev** [reply]
Could you paste your collections configuration as json (_Admin UI > Settings > Export collections_)?

**Comment by dimitarnikolovv** [comment]
[pb_schema.json](https://github.com/pocketbase/pocketbase/files/13299681/pb_schema.json)

**Reply by ganigeorgiev** [reply]
It works fine for me when testing with PocketBase v0.19.3. You can try to access the list url directly in the browser:
```
http://localhost:8090/api/collections/posts/records?expand=author.ratings
```

<details>
  <summary>Click to expand the sample JSON response that it is returned</summary>

```json
{
  "page": 1,
  "perPage": 30,
  "totalItems": 1,
  "totalPages": 1,
  "items": [
    {
      "additionalInformation": "",
      "additionalStops": null,
      "arriveAt": "asdasd",
      "author": "xxmvdmmksy5dm8x",
      "car": "",
      "collectionId": "k8oe3hlpvmqbb7v",
      "collectionName": "posts",
      "created": "2023-11-08 17:43:36.286Z",
      "departDate": "2023-11-04 12:00:00.000Z",
      "departDateTime": "2023-11-01 12:00:00.000Z",
      "departFrom": "asdasd",
      "departTime": "asdasd",
      "departTimeSecond": "",
      "expand": {
        "author": {
          "age": 12,
          "avatar": "v0_11_0_s0_sn_i9j_wea_Puyzcx0YS2.png",
          "collectionId": "_pb_users_auth_",
          "collectionName": "users",
          "created": "2023-10-29 13:12:24.190Z",
          "emailVisibility": false,
          "expand": {
            "ratings": [
              {
                "author": "",
                "collectionId": "nsk0enxbx3gh78p",
                "collectionName": "ratings",
                "comment": "",
                "created": "2023-11-08 17:43:55.608Z",
                "id": "9tg6doguy2d0u2r",
                "rating": 1,
                "target": "",
                "updated": "2023-11-08 17:43:55.608Z"
              },
              {
                "author": "",
                "collectionId": "nsk0enxbx3gh78p",
                "collectionName": "ratings",
                "comment": "",
                "created": "2023-11-08 17:43:59.994Z",
                "id": "9dwdetz62vdc9ev",
                "rating": 2,
                "target": "",
                "updated": "2023-11-08 17:43:59.994Z"
              },
              {
                "author": "",
                "collectionId": "nsk0enxbx3gh78p",
                "collectionName": "ratings",
                "comment": "",
                "created": "2023-11-08 17:44:03.240Z",
                "id": "6pdg8qlxz6cgk5t",
                "rating": 3,
                "target": "",
                "updated": "2023-11-08 17:44:03.240Z"
              }
            ]
          },
          "firstName": "asdasd",
          "gender": "male",
          "id": "xxmvdmmksy5dm8x",
          "lastName": "adsasd",
          "nameVisibility": "",
          "ratings": [
            "9tg6doguy2d0u2r",
            "9dwdetz62vdc9ev",
            "6pdg8qlxz6cgk5t"
          ],
          "showAvatar": false,
          "showTel": false,
          "updated": "2023-11-08 17:44:06.736Z",
          "username": "users86631",
          "verified": false
        }
      },
      "id": "r7r8kdzx5tf5ztp",
      "passengers": 0,
      "price": 0,
      "tel": "1231231232",
      "updated": "2023-11-08 17:44:54.738Z"
    }
  ]
}
```
</details>

**Reply by dimitarnikolovv** [reply]
I can confirm that It works In the production database when accessed through the API ([you can check](https://backend.sharetravel.bg/api/collections/posts/records?expand=author.ratings)), but when I try to get the data in my backend it simply does not include the author's expand field.

**Reply by ganigeorgiev** [reply]
I'm not sure how to help you debug it.

If you use Next.js or other similar frontend framework, make sure that there is no cache involved.

If you think that this is a bug with PocketBase, please create a **minimal** reproducable repo including test `pb_data` and I'll investigate it further.

**Reply by dimitarnikolovv** [reply, marked as answer]
It was a cache-related issue, a colleague of mine has implemented a caching system that I wasn't aware of. Thanks.
